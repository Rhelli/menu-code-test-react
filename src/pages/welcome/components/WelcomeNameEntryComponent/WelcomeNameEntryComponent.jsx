import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './WelcomeNameEntryComponent.module.scss';

const WelcomeNameEntryComponent = ({ submitNewOrders }) => {
  const [guest1, setGuest1] = useState(null);
  const [guest2, setGuest2] = useState(null);

  return (
    <div className={styles.welcomeNameEntryFormComponent}>
      <h3>Enter Guest Names</h3>
      <form onSubmit={(event) => submitNewOrders(event, guest1, guest2)} className={styles.nameEntryForm}>
        <div
          onChange={(event) => setGuest1(event.target.value)}
          className={styles.nameEntry}
        >
          <label>
            <h4>Guest One&apos;s Name</h4>
            <input id="guest1" type="text" placeholder="Optional" />
          </label>
        </div>
        <div
          onChange={(event) => setGuest2(event.target.value)}
          className={styles.nameEntry}
        >
          <label>
            <h4>Guest Two&apos;s Name</h4>
            <input id="guest2" type="text" placeholder="Optional" />
          </label>
        </div>
        <button className={styles.nameEntryButton} type="submit">
          Continue
      </button>
      </form>
    </div>
  );
};

WelcomeNameEntryComponent.propTypes = {
  submitNewOrders: PropTypes.func.isRequired
};

export default WelcomeNameEntryComponent;
