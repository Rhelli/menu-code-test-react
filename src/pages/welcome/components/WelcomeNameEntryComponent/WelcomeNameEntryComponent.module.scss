.welcomeNameEntryFormComponent {
  width: 100%;
  height: auto;
  padding: 20px 0;

  h3 {
    width: 100%;
    text-align: center;
    font-size: 24px;
    color: #202020;
    border-bottom: 1px solid #d8d9db;
    padding-bottom: 14px;
  }

  .nameEntryForm {
    display: flex;
    flex-flow: column nowrap;
    width: 100%;
    border-bottom: 1px solid #d8d9db;
    padding: 20px 0;

    .nameEntry {
      width: 100%;
      padding: 10px 0;

      h4 {
        color: #404040;
        font-size: 14px;
      }

      label {
        width: 100%;

        input {
          width: 100%;
          height: 35px;
          outline: none;
          border: none;
          border-bottom: 1px solid #d8d9db;
          transition: 0.1s;

          &::placeholder {
            font-family: inherit;
            font-size: 12px;
          }

          &:hover,
          &:focus {
            transition: 0.1s;
            border-bottom: 2px solid #da3743;
          }
        }
      }
    }

    button {
      align-self: center;
      width: 55%;
      height: 45px;
      outline: none;
      border: none;
      font-size: 16px;
      background-color: #da3743;
      color: #fff;
      font-family: inherit;
      border-radius: 2px;
      transition-duration: 0.1s;

      &:hover,
      &:focus {
        background-color: #DF505B;
        transition-duration: 0.1s;
      }
    }
  }
}