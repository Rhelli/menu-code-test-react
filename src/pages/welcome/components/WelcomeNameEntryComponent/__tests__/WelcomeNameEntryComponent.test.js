import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import WelcomeNameEntryComponent from '../WelcomeNameEntryComponent';

Enzyme.configure({ adapter: new Adapter() });

const submitNewOrders = jest.fn(() => 'New order submitted');
const welcomeNameEntryComponent = (<WelcomeNameEntryComponent submitNewOrders={submitNewOrders} />);

describe('The appearance and functionality of the WelcomeNameEntryComponent', () => {
  const container = mount(<WelcomeNameEntryComponent submitNewOrders={submitNewOrders} />);
  it('Matches the previous snapshot', () => {
    const tree = renderer.create(welcomeNameEntryComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly', () => {
    window.HTMLCanvasElement.prototype.getContext = () => {};
    mount(<WelcomeNameEntryComponent submitNewOrders={submitNewOrders} />);
  });

  it('Contains the form element with the correct attributes', () => {
    expect(container.find('.nameEntryForm').name()).toBe('form');
  });

  it('Contains two entry points for names', () => {
    expect(container.find('#guest1')).toBeTruthy();
    expect(container.find('#guest2')).toBeTruthy();
  });
});
