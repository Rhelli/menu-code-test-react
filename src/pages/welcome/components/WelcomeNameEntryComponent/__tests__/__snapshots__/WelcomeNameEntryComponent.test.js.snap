// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the WelcomeNameEntryComponent Matches the previous snapshot 1`] = `
<div
  className="welcomeNameEntryFormComponent"
>
  <h3>
    Enter Guest Names
  </h3>
  <form
    className="nameEntryForm"
    onSubmit={[Function]}
  >
    <div
      className="nameEntry"
      onChange={[Function]}
    >
      <label>
        <h4>
          Guest One's Name
        </h4>
        <input
          id="guest1"
          placeholder="Optional"
          type="text"
        />
      </label>
    </div>
    <div
      className="nameEntry"
      onChange={[Function]}
    >
      <label>
        <h4>
          Guest Two's Name
        </h4>
        <input
          id="guest2"
          placeholder="Optional"
          type="text"
        />
      </label>
    </div>
    <button
      className="nameEntryButton"
      type="submit"
    >
      Continue
    </button>
  </form>
</div>
`;
